rotkehlchen/db/taxable_events.py

```diff
@@ -0,0 +1,348 @@
+    :param event:
+    :return: The bytes representation of the event
+    """
+    return pickle.dumps(event)
+
+
+def deserialize_event_from_db(result: bytes) -> Dict[str, Any]:
+    """
+    Deserialize the event as it was stored in the database.
+
+    :param result: A SELECT query result; specifically a bytes blob contained in the data column
+    :return event: The typed event
+    """
+    return pickle.loads(result)
+
+
+def _return_reports_or_events_maybe_limit(
+        entry_type: str,
+        entries: List[Dict[str, Any]],
+        with_limit: bool,
+) -> List[Dict[str, Any]]:
+    if with_limit is False:
+        return entries
+
+    if entry_type == 'events':
+        limit = FREE_PNL_EVENTS_LIMIT
+    else:
+        limit = FREE_REPORTS_LOOKUP_LIMIT
+
+    count: int = 0
+    for _ in entries:
+        count += 1
+
+    returning_entries_length = min(limit, len(entries))
+
+    return entries[:returning_entries_length]
+
+
+class DBTaxableEvents(LockableQueryMixIn):
+
+    def __init__(self,
+                 database: 'DBHandler',
+                 msg_aggregator: MessagesAggregator):
+        super().__init__()
+        self.db = database
+        self.msg_aggregator = msg_aggregator
+
+    @protect_with_lock()
+    def query(
+            self,
+            filter_query: ReportsFilterQuery,
+            with_limit: bool = False,
+            only_cache: bool = False,
+    ) -> List[Dict[str, Any]]:
+        """Queries for all reports/events of a Profit and Loss Report which has been performed
+        historically. Returns a list of all transactions filtered and sorted according to the
+        parameters.
+
+        If `with_limit` is true then the api limit is applied
+
+        if `recent_first` is true then the transactions are returned with the most
+        recent first on the list
+
+        May raise:
+        - TODO check what it may raise
+        """
+        report_id = filter_query.report_id or None
+
+        if only_cache is False:
+            # f_from_ts = filter_query.from_ts
+            # f_to_ts = filter_query.to_ts
+            # from_ts = Timestamp(0) if f_from_ts is None else f_from_ts
+            # to_ts = ts_now() if f_to_ts is None else f_to_ts
+            if report_id is not None:
+                # TODO: only_cache was False, we need to hand off
+                #  what we have and generate the rest
+                pass
+
+        if report_id is None:
+            entries = self.get_reports(filter_=filter_query)
+        else:
+            report = self.get_reports(filter_=filter_query)[0]
+            events_filter_query = filter_query
+            events_filter_query.report_id_filter = DBEventsReportIDFilter(and_op=True,
+                                                                          report_id=report_id)
+            entries = self.get_events(filter_=events_filter_query)
+        entry_type = 'reports' if report_id is None else 'events'
```

Comment:
In such places you need a newline to easily determine where the next section starts

```suggestion
            entries = self.get_events(filter_=events_filter_query)

        entry_type = 'reports' if report_id is None else 'events'
```